src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -77,6 +79,58 @@ internal static ParseResult ParseScript(string script, ILanguageSettings loc = n
             return new ParseResult(parsetree, errors, errors?.Any() ?? false, parser._comments, parser._before, parser._after);
         }
 
+        public static ParseFormulasResult ParseFormulasScript(string script, ILanguageSettings loc = null, Flags flags = Flags.None)
+        {
+            Contracts.AssertValue(script);
+            Contracts.AssertValueOrNull(loc);
+
+            Token[] formulaTokens = TokenizeScript(script, loc, flags);
+            TexlParser parser = new TexlParser(formulaTokens, Flags.NamedFormulas);
```

Comment:
Flags.NamedFormulas | flags (combine with the flags argument)

maybe set that as a variable and pass to both TokenizeScript and TexlParser, just in case we want the namedformula flag in both later.